msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md

```diff
@@ -0,0 +1,310 @@
+> [!div class="checklist"]
+> [!div class="checklist"]
+>
+> * **id**  — your Azure AD app id.
+> * **resource** — the resource URL for the app.
+>
+
+>[!NOTE]
+>
+> * Your bot requires _application_ not _user delegated_ permissions because the installation is not for yourself but for others.
+>
+> * An Azure AD tenant administrator must [explicitly grant permissions to an application](/graph/security-authorization#grant-permissions-to-an-application). After an application is granted permissions, _all_ members of the Azure AD tenant will gain the granted permissions.
+
+## Enable proactive app installation and messaging
+
+ > [!IMPORTANT]
+>Microsoft Graph will only install apps published within your organization's [app catalog](../../concepts/deploy-and-publish/overview.md#publish-to-your-organizations-app-catalog) or in [AppSource](https://appsource.microsoft.com/).
+
+### ✔ Create and publish your proactive messaging bot for Teams
+
+To get started, you will need a [bot for Teams](../../bots/how-to/create-a-bot-for-teams.md) with [proactive messaging](../../concepts/bots/bot-conversations/bots-conv-proactive.md) capabilities and  [published](../../concepts/deploy-and-publish/overview.md) in your organization's [app catalog](../../concepts/deploy-and-publish/overview.md#publish-to-your-organizations-app-catalog) or in [AppSource](https://appsource.microsoft.com/).
+
+>[!TIP]
+> The production-ready [**Company Communicator**](../..//samples/app-templates.md#company-communicator) app template enables broadcast messaging and is a good foundation for building your proactive bot application.
+
+### ✔ Get the `teamsAppId` for your app
+
+You will need the `teamsAppId`  for the next steps.
+
+1. The `teamsAppId` can be retrieved from your organization's app catalog:
+
+**Microsoft Graph page reference:** [teamsApp resource type](/graph/api/resources/teamsapp?view=graph-rest-1.0)
+
+**HTTP GET** request:
+
+```http
+GET /appCatalogs/teamsApps?$filter=externalId eq '{IdFromManifest}'
+```
+
+The request will return a `teamsApp`  object. The returned object's `id`  is the app's catalog generated app ID and is different from the "id:" that you provided in your Teams app manifest:
+
+```json
+{
+  "value": [
+    {
+      "id": "b1c5353a-7aca-41b3-830f-27d5218fe0e5",
+      "externalId": "f31b1263-ba99-435a-a679-911d24850d7c",
+      "name": "Test App",
+      "version": "1.0.1",
+      "distributionMethod": "Organization"
+    }
+  ]
+}
+```
+
+2. If your app has already been uploaded/sideloaded for a user in the personal scope, you can retrieve the `teamsAppId` as follows:
+
+**Microsoft Graph page reference:** [List apps installed for user](/graph/api/user-list-teamsappinstallation?view=graph-rest-beta&tabs=http)
+
+**HTTP GET** request:
+
+```http
+GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}'
```

Comment:
In this case, the user is trying to find the teamsAppId, so the query should be:

https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/externalId eq '{manifestId}'
```